apps/bot/db/actions/users.ts

```diff
@@ -143,3 +143,17 @@ export const removePointsFromUser = async (
   type: keyof typeof POINTS_REWARDS,
   trx: TransactionDB | KyselyDB = db,
 ) => updatePointsBySum(userId, -POINTS_REWARDS[type], trx)
+
+export const getCorrectAnswersCount = (userId: string) => {
+  return db
+    .selectFrom('users')
+    .where('snowflakeId', '=', userId)
+    .select(['answersCount'])
+    .executeTakeFirst()
+}
+
+export const isUserProfilePublic = async (user: GuildMember) => {
+  if (env.PUBLIC_PROFILE_ROLE_ID)
+    return !user.roles.cache.has(env.PUBLIC_PROFILE_ROLE_ID)
```

Comment:
ngl it feels weird to have the return type of "is user public profile" to be the opposite.

also, i dont think this function needs to be async and even in this file (as not doing anything with db)

and random thought, can you be weird and `return env.PUBLIC_PROFILE_ROLE_ID && user.roles.cache.has(env.PUBLIC_PROFILE_ROLE_ID)`